app/src/main/java/com/sopt/now/compose/feature/auth/SignUpActivity.kt

```diff
@@ -13,153 +16,77 @@ import androidx.compose.material3.MaterialTheme
-    private fun checkMbtiLength(
-        user: User,
-        validationMessages: MutableList<String>
-    ) {
-        if (user.mbti.length != MBTI_LENGTH) {
-            validationMessages.add(getString(R.string.sign_up_mbti_snack_message, MBTI_LENGTH))
-        }
-    }
-
-    private fun checkNickNameLength(
-        user: User,
-        validationMessages: MutableList<String>
-    ) {
-        if (user.nickName.length <= MIN_NICKNAME_LENGTH) {
-            validationMessages.add(
-                getString(
-                    R.string.sign_up_nickname_snack_message,
-                    MIN_NICKNAME_LENGTH
-                )
-            )
-        }
-    }
-
-    private fun validateLength(
-        input: String,
-        minLength: Int,
-        maxLength: Int,
-        fieldNameResId: Int
-    ): String? {
-        return if (input.length !in minLength..maxLength) {
-            getString(
-                R.string.sign_up_snack_message,
-                getString(fieldNameResId),
-                minLength,
-                maxLength
-            )
-        } else {
-            null
-        }
-    }
-
-    private fun handleValidationResult(validationMessages: List<String>, userInputData: User) {
-        if (validationMessages.isEmpty()) {
-            navigateToLoginActivity(userInputData)
-            toast("회원가입이 완료되었습니다.")
-        } else {
-            toast(validationMessages.joinToString("\n"))
-        }
-    }
-
-    private fun navigateToLoginActivity(userInputData: User) {
-        Intent().apply {
-            putExtra(KeyStorage.USER_INPUT, userInputData)
-            setResult(RESULT_OK, this)
-            finish()
-        }
-    }
-
-    companion object {
-        const val MIN_ID_LENGTH = 6
-        const val MAX_ID_LENGTH = 10
-        const val MIN_PWD_LENGTH = 8
-        const val MAX_PWD_LENGTH = 12
-        const val MIN_NICKNAME_LENGTH = 2
-        const val MBTI_LENGTH = 4
-    }
 }
 
 @Composable
 fun SignUpScreen(
-    onSignUpBtnClick: (User) -> Unit
+    onSignUpBtnClick: (User) -> Unit,
+    viewModel: SignUpViewModel
 ) {
     var id by remember { mutableStateOf("") }
     var password by remember { mutableStateOf("") }
     var nickName by remember { mutableStateOf("") }
     var mbti by remember { mutableStateOf("") }
+
+    val context = LocalContext.current
+    val signUpState by viewModel.signUpResponseState.collectAsState(initial = UiState.Loading)
+
+    LaunchedEffect(signUpState) {
+        viewModel.signUpResponseState.collect { uiState ->
+            when (uiState) {
+                is UiState.Success -> {
+                    Toast.makeText(context, uiState.data.toString(), Toast.LENGTH_SHORT).show()
+                    navigateToLoginActivity(context, uiState.data)
+                }
+
+                is UiState.Failure -> {
+                    Toast.makeText(context, uiState.errorMessage, Toast.LENGTH_SHORT).show()
+                }
+
+                else -> Unit
+            }
```

Comment:
[P1] 
LoginActivity에서와 마찬가지로 collect는 한 번만 수행해도 괜찮을 것 같습니다.

```suggestion
    LaunchedEffect(Unit) {
        viewModel.signUpResponseState.collect { uiState ->
            when (uiState) {
                is UiState.Success -> {
                    Toast.makeText(context, uiState.data.toString(), Toast.LENGTH_SHORT).show()
                    navigateToLoginActivity(context, uiState.data)
                }

                is UiState.Failure -> {
                    Toast.makeText(context, uiState.errorMessage, Toast.LENGTH_SHORT).show()
                }

                else -> Unit
            }
```

Reply:
[f73a049](https://github.com/NOW-SOPT-ANDROID/chanwoo-bae/pull/17/commits/f73a04900df19745f70149cdee8e57b6539e7b77) 감사합니다! 수정했습니당